packages/safe-apps-provider/src/provider.ts

```diff
@@ -195,6 +195,90 @@ export class SafeAppProvider extends EventEmitter implements EIP1193Provider {
       case 'safe_setSettings':
         return this.sdk.eth.setSafeSettings([params[0]]);
 
+      case 'wallet_sendCalls': {
+        if (params[0].from !== this.safe.safeAddress) {
+          throw Error('Invalid from address');
+        }
+
+        const txs = params[0].calls.map(
+          (call: { to?: `0x${string}`; data?: `0x${string}`; value?: `0x${string}`; chainId?: `0x${string}` }) => {
+            if (call.chainId !== numberToHex(this.chainId) || !call.to) {
+              throw new Error('Invalid call');
```

Comment:
We should maybe give more details: e.g. "Invalid calldata: The Safe is not available on the chain"

Reply:
I improved the error messages in c3ab8181e8a18c52b1f5fbaa13f78e1e5ec0b292. What do you think?